boards/Makefile.common

```diff
@@ -247,9 +247,17 @@ target:
 %.elf: %
 	$(Q)cp $< $@
 
-%.bin: %.elf $(TOCK_ROOT_DIRECTORY)tools/sha256sum/target/debug/sha256sum
+# Check whether the system already has a sha256sum application
+# present, if not use the custom shipped one
+ifeq (, $(shell which sha256sum))
+  %.bin: %.elf $(TOCK_ROOT_DIRECTORY)tools/sha256sum/target/debug/sha256sum
 	$(Q)$(OBJCOPY) --output-target=binary $(OBJCOPY_FLAGS) $< $@
 	$(Q)$(TOCK_ROOT_DIRECTORY)tools/sha256sum/target/debug/sha256sum $@
+else
+  %.bin: %.elf
+	$(Q)$(OBJCOPY) --output-target=binary $(OBJCOPY_FLAGS) $< $@
+	$(Q)sha256sum $@
+endif
```

Comment:
Instead of having the ifdef here, I think it'd be better to do what we do elsewhere through the Tock make system and have a check at the start that sets a `SHA256SUM` variable, which is then used unconditionally in the rules. It makes it a bit easier to read to separate the 'configuration' from the 'build rules'.

The other small change is a light preference for the built-in `command -v` rather than `which`; I no longer recall when and why this was a problem, but I remember that it was once.

So this would be something more like

```make
ifeq (,$(shell command -v sha256sum)
  # No system sha256 available
  SHA256SUM := ....
else
   # Use system sha256
  SHA256SUM := sha256sum
endif


...

rule:
     $(Q)$(SHA256SUM) $@
```

Reply:
Thanks for the hints! I did actually try that approach, but must've been doing something wrong as it didn't work and the `make` errors were not particularly helpful. I'll apply the changes requested.

Reply:
Thanks for the hints! I actually tried that approach at first, but must've been doing something wrong and the `make` errors were not particularly helpful. I'll apply the changes requested.